packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts

```diff
@@ -496,177 +496,187 @@ export default class TableResize implements EditorPlugin {
+                                const newHeight = originalHeight * ratioY;
+                                if (newHeight >= MIN_CELL_HEIGHT) {
+                                    cell.td.style.height = `${newHeight}px`;
                                 }
+                            } else {
+                                cell.td.style.height = null;
                             }
                         }
                     }
                 }
             }
-            rect = normalizeRect(this.currentTable.getBoundingClientRect());
-            currentBorder = this.isRTL ? rect.left : rect.right;
-            this.currentTable.setAttribute(
-                this.isRTL ? 'currentLeftBorder' : 'currentRightBorder',
-                currentBorder.toString()
-            );
+        }
+        rect = normalizeRect(vtable.table.getBoundingClientRect());
+        currentBorder = this.isRTL ? rect.left : rect.right;
+        vtable.table.setAttribute(
+            this.isRTL ? 'currentLeftBorder' : 'currentRightBorder',
+            currentBorder.toString()
+        );
 
-            const currentBottomBorder = this.currentTable.getBoundingClientRect().bottom;
-            this.currentTable.setAttribute('currentBottomBorder', currentBottomBorder.toString());
-            vtable.writeBack();
+        const currentBottomBorder = vtable.table.getBoundingClientRect().bottom;
+        vtable.table.setAttribute('currentBottomBorder', currentBottomBorder.toString());
+        vtable.writeBack();
+        return;
+    };
+
+    private resizeRows = (newPos: number, rect: Rect, vtable: VTable) => {
+        if (!this.isReadyForResizingHeight) {
+            preProcessTable(vtable.table, ResizeState.Horizontal);
+            this.isReadyForResizingHeight = true;
+        }
+        vtable.forEachCellOfCurrentRow(cell => {
+            if (cell.td) {
+                cell.td.style.height = cell.td == this.currentTd ? `${newPos - rect.top}px` : null;
+            }
+        });
+    };
+
+    private resizeColumns = (newPos: number, vtable: VTable, isShiftPressed: boolean) => {
+        if (!this.isReadyForResizingWidth) {
+            preProcessTable(vtable.table, ResizeState.Vertical);
+            this.isReadyForResizingWidth = true;
+        }
+        // Since we allow the user to resize the table width on adjusting the border of the last cell,
+        // we need to make the table width resizeable by setting it as null;
+        // We also allow the user to resize the table width if Shift key is pressed
+        if (!this.currentCellsToResize[0].nextElementSibling || isShiftPressed) {
+            vtable.table.removeAttribute('width');
+            vtable.table.style.width = null;
+        }
+
+        this.currentCellsToResize.forEach(td => {
+            const rect = normalizeRect(td.getBoundingClientRect());
+            const nextTd = td.nextElementSibling as HTMLTableElement;
```

Comment:
Use VTable to help you get next TD. Because next Element sibling may not be TD.

Reply:
Looks like we don't have existing method in VTable to get the next TD, we have Virtual cells there, do you mean we return the cells for the method getCellsWithBorder, and also create a method to get the next cells?

Reply:
For example, if you want to know if a cell is the last cell of this row, you can do:

const isLastTd = td == vTable.getTd(rowIndex, vTable.cells[rowIndex].length - 1)  // Need to amke getTd public

Reply:
And to get nextTd for resize column, you can do 
vTable.getCell(rowIndex, colIndex + 1)?.td

If it return null, it means that is a merged cell so no need to resize.

Reply:
fixed